common/Toolbox.ts

```diff
@@ -34,6 +35,52 @@ export function concatenatedStringRegex(strings: string[]) {
   return new RegExp(`\\b(${allStrings.join("|")})\\b`, "gim");
 }
 
+/**
+ * This is just a wacky experiment in speeding something up that isn't even slow. :D
+ */
+class StringTrie {
```

Comment:
Cool idea! Consider putting this in its own package, but I don't think it's needed here.

Reply:
Yep forgot to prune that.